packages/dds/merge-tree/src/client.ts

```diff
@@ -706,11 +706,14 @@ export class Client {
                 case MergeTreeDeltaType.ANNOTATE:
                     assert(segment.propertyManager?.hasPendingProperties() === true,
                         0x036 /* "Segment has no pending properties" */);
-                    newOp = createAnnotateRangeOp(
-                        segmentPosition,
-                        segmentPosition + segment.cachedLength,
-                        resetOp.props,
-                        resetOp.combiningOp);
+                    // if the segment has been removed, there's no need to send the annotate op
+                    if (segment.removedSeq === undefined && segment.localRemovedSeq === undefined) {
```

Comment:
i don't think this condition is quite correct. as we want to regenerate when `localRemovedSeq !== undefined` and in that case `removedSeq` will also not be `undefined`